lib/shared/errors.js

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+
+/**
+ * @summary Human-friendly error messages
+ * @namespace HUMAN_FRIENDLY
+ * @public
+ */
+exports.HUMAN_FRIENDLY = {
+
+  /* eslint-disable new-cap */
+
+  /**
+   * @namespace ENOENT
+   * @memberof HUMAN_FRIENDLY
+   */
+  ENOENT: {
+
+    /**
+     * @property {Function} title
+     * @memberof ENOENT
+     * @param {Error} error - error object
+     * @returns {String} message
+     */
+    title: (error) => {
+      return `No such file or directory: ${error.path}`;
+    },
+
+    /**
+     * @property {Function} description
+     * @memberof ENOENT
+     * @param {Error} error - error object
+     * @returns {String} message
+     */
+    description: _.constant('The file you\'re trying to access doesn\'t exist')
```

Comment:
Sorry for the belated response – I was sure I commented last week, but must have screwed it up or forgotten.

> I guess we could make errors that don't need such feature strings, and check the type in the helper function. What do you think?

I think we can keep it as it is, but to cut down on the verbosity of this, what would you think about adding a class / function which handles the title / description setting?

```js
class ErrorDetails {
  constructor( options ) {
    
    /**
     * @property {Function} title
     * @param {Error} error - error object
     * @returns {String} message
     */
    this.title = typeof options.title !== 'function'
      ? _.constant( options.title )
      : options.title
    
    /**
     * @property {Function} description
     * @param {Error} error - error object
     * @returns {String} message
     */
    this.description = typeof options.description !== 'function'
      ? _.constant( options.description )
      : options.description
    
  }
}
```

Which would reduce declaring errors to:

```js
ENOENT: new ErrorDetails({
  title( error ) {
    return `No such file or directory: ${error.path}`
  },
  description: 'The file you\'re trying to access doesn\'t exist'
})
```